### Hanan Sherka 
### Statistical Analysis of Lottery Aid to Education in New York With PANDAS
### 09.25.18

## Abstract

In this lab, I took data on New York's distribution of lottery money towards school aid in the years 2002 - 2017. From this, I looked for and found the most aid given, the least aid given, average aid given, median aid given to each school district, and the difference between the minimum and maximum amount of aid given. I got this data from data.gov, I know it is reputable because it is a government website. This is the link to the untouched data set https://data.ny.gov/api/views/9ypc-vjig/rows.csv . 

## Dataset Preparation 

I opened the data file of Lottery Aid to Education Data and made it a dataframe and data structure. 

In [35]:
#always include aliased packages
import math as m 
import numpy as np
import scipy as sp 
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt 
import seaborn as sns


In [36]:
#one line to open and read a csv file into a date frame data structure, and replace the header names 
headernames = ["bfy", "efy","county","school_district", "aid"]
df= pd.read_csv("LATE_original.csv", names = headernames, header = 0)

In [37]:
# make it so that all the county names are the same case so they are easier to work with/ recognized by county
df['county'] = df['county'].str.lower()
df.head ()

bfy   efy             county      school_district         aid
0  2002  2003  albany                          Albany  7242297.04
1  2002  2003  albany             Berne-Knox-Westerlo   874046.94
2  2002  2003  albany                       Bethlehem  2786954.71
3  2002  2003  albany                          Cohoes  1820297.94
4  2002  2003  albany                    Green Island   241859.93

In [38]:
#Test to veryify that it would print the first four values in the list
print(df.bfy[:4])
print(df.efy[:4]) 
print(df.county[:4])
print(df.school_district[:4])
print(df.aid[:4])

0    2002
1    2002
2    2002
3    2002
Name: bfy, dtype: int64
0    2003
1    2003
2    2003
3    2003
Name: efy, dtype: int64
0    albany           
1    albany           
2    albany           
3    albany           
Name: county, dtype: object
0                 Albany
1    Berne-Knox-Westerlo
2              Bethlehem
3                 Cohoes
Name: school_district, dtype: object
0    7242297.04
1     874046.94
2    2786954.71
3    1820297.94
Name: aid, dtype: float64


## Data Modeling

I used the numpy package to find the mean, median, max, and min aid given. 

In [39]:
#use mean method from numpy to find the average aid given 
aidmean = np.mean(df.aid) 

print(aidmean)

4063464.2532175635


In [40]:
#use mean method from numpy to find the min and max aid given 
maxaid = np.max(df.aid)
minaid = np.min(df.aid)

print(maxaid)
print(minaid)

1337283719.0
105.0


In [44]:
#use the median method from numpy to find the median aid given 
medianaid = np.median(df.aid) 
print(medianaid)

1474788.94


In [55]:
#use the subtract method from numpy to find the difference between the max and min aid
difference = np.subtract(maxaid, minaid)
print(difference)

1337283614.0


## Data Analysis and Conclusion 

 I found that the most aid given to a school district in New York State from lottery funds was \$1,337,283,719, and the least aid given to a shcool district was \$105. I did not know how to index a dataframe, so I could not attach these numbers to the specific school district that recieved these amounts. The average aid given was \$4,063,464.25. Two new things I was able to discover in this lab using the numpy package were the median aid, and the difference of aid given between the district that recieved the most and the district that recieved the least aid. The median aid given was \$1337283719, and the difference between the max and min aid given to any school district was \$1337283614. This information made me wonder how a school district could make \$105 when the average given is millions of dollars more than that. It also made me wonder how the money given to districts from the lottery is distributed. These are questions up for future investigation. 


## Acknowledgements

I would like to acknowledge Ms. Sconyers for providing code on how to use the numpy package on the Pandas Practice file. Also, to this website http://cs231n.github.io/python-numpy-tutorial/#numpy-array-indexing for giving a rundown on how to use all the packages in python we have been introduced to so far, and showing many things I could do with these packages. 